pkgs/applications/networking/browsers/waterfox/default.nix

```
@@ -0,0 +1,212 @@
+{ lib
+, stdenv
+, fetchurl
+, wrapGAppsHook
+, alsa-lib
+, at-spi2-atk
+, at-spi2-core
+, atk
+, cairo
+, cups
+, dbus-glib
+, libnotify
+, libdrm
+, libxkbcommon
+, mesa
+, nspr
+, nss
+, pango
+, systemd
+, libpulseaudio
+, gdk-pixbuf
+, gtk3
+, makeDesktopItem
+, unzip
+, libX11
+, libXScrnSaver
+, libXcomposite
+, libXcursor
+, libXdamage
+, libXext
+, libXfixes
+, libXi
+, libXrandr
+, libXrender
+, libXtst
+, libxcb
+, writeText
+, patchelf
+, makeWrapper
+, libGL
+, ffmpeg
+, glib
+, pciutils  # Added for libpci.so
+}:
+
+let
+  version = "6.5.4";
+  
+  desktopItem = makeDesktopItem {
+    name = "waterfox";
+    exec = "waterfox %U";
+    icon = "waterfox";
+    desktopName = "Waterfox";
+    genericName = "Web Browser";
+    categories = [ "Network" "WebBrowser" ];
+    mimeTypes = [
+      "text/html"
+      "text/xml"
+      "application/xhtml+xml"
+      "application/vnd.mozilla.xul+xml"
+      "x-scheme-handler/http"
+      "x-scheme-handler/https"
+      "x-scheme-handler/ftp"
+    ];
+  };
+  
+  prefsFile = writeText "waterfox-prefs.js" ''
+    // Disable automatic updates and notifications
+    pref("app.update.auto", false);
+    pref("app.update.enabled", false);
+    pref("browser.crashReports.unsubmittedCheck.enabled", false);
+  '';
+
+in stdenv.mkDerivation rec {
+  pname = "waterfox";
+  inherit version;
+
+  src = fetchurl {
+    url = "https://cdn1.waterfox.net/waterfox/releases/${version}/Linux_x86_64/waterfox-${version}.tar.bz2";
+    sha256 = "d73a14315199a88d91b42b6d7f8dd6d427962d8a10d23c6717ac749f42292d0d";
+  };
+
+  nativeBuildInputs = [
+    wrapGAppsHook
+    unzip
+    patchelf
+    makeWrapper
+  ];
+
+  buildInputs = [
+    alsa-lib
+    at-spi2-atk
+    at-spi2-core
+    atk
+    cairo
+    cups
+    dbus-glib
+    libnotify
+    libdrm
+    libxkbcommon
+    mesa
+    nspr
+    nss
+    pango
+    systemd
+    libpulseaudio
+    gdk-pixbuf
+    gtk3
+    libX11
+    libXScrnSaver
+    libXcomposite
+    libXcursor
+    libXdamage
+    libXext
+    libXfixes
+    libXi
+    libXrandr
+    libXrender
+    libXtst
+    libxcb
+    stdenv.cc.cc.lib
+    libGL
+    ffmpeg
+    glib
+    pciutils  # Added here
+  ];
+
+  dontBuild = true;
+  dontConfigure = true;
```

```suggestion
  dontConfigure = true;
  dontBuild = true;
```
Configure phase precede build